core/src/main/scala/org/apache/spark/shuffle/sort/SortShuffleManager.scala

```diff
@@ -254,10 +256,18 @@ private[spark] object SortShuffleManager extends Logging {
     val executorComponents = ShuffleDataIOUtils.loadShuffleDataIO(conf).executor()
     val extraConfigs = conf.getAllWithPrefix(ShuffleDataIOUtils.SHUFFLE_SPARK_CONF_PREFIX)
         .toMap
+    val pluginOutputTrackingEnabled = conf.get(config.SHUFFLE_IO_PLUGIN_OUTPUT_TRACKING_ENABLED)
+    val trackerWorker = SparkEnv.get.mapOutputTracker.asInstanceOf[MapOutputTrackerWorker]
```

Comment:
Could it be better to have updater as a component rather than making MapOutputTrackerWorker extend it.. ??
`val shuffleUpdater = SparkEnv.get.shuffleUpdater `. something like this?

Because i think we should not change the MapOutputTrackerWorker's model of working..

Reply:
Component is probably better, yeah.